### DISCUSSION TITLE
`ros2 topic list` shows partial list of topics

### DISCUSSION THREAD

**Discussion by scottcandy34** [iRobotEducation/create3_docs#359] [Questions]
I installed the H.1.0 firmware and there is missing lots of commands for ros2. When running `ros2 topic list` I only get this returned (below) which is very different from H.0.0 and G.4.1. its essentially missing important topics.

```
/cmd_audio
/cmd_lightring
/cmd_vel
/interface_buttons
/mobility_monitor/transition_event
/parameter_events
/robot_state/transition_event
/rosout
/static_transform/transition_event
```

I am not sure if this was the intended outcome or not but since there was no changelog on the docs page for the firmware release saying that it was changed I made this issue (in fact it says it did not change the ROS API commands). I discovered this issue when trying to control the robot with your create3_examples teleop commands and the robot not moving. Changing to H.0.0 or back to any galactic version works fine.

**Comment by alsora** [comment]
Hi @scottcandy34,
this seems to be a connectivity issue between your laptop and your robot.

Try running `ros2 topic list` multiple times or even better `ros2 topic list --no-daemon --spin-time 10` and this should show the full list of topics.

P.S. no topics have been removed or changed in H.1.0

**Reply by scottcandy34** [reply]
No its the PI over USB C and it works fine for any other firmware version (not a single issue). I have gone back and fourth between the different versions which work but each time I install firmware H.1.0 it has issues and does not work correctly. Running the `ros2 topic list` multiple times does not change a thing. 

I know there have been no changes that's why it is strange to see it not work. This is an issue on the iRobot itself, from my diagnostic testing, something on that particular update changed something. 

Can you please check the current H.1.0 yourself.

**Reply by shamlian** [reply]
Just to double check, and so we can try to reproduce this: what middleware are you using on your robot and your Pi?

**Reply by scottcandy34** [reply]
I am using ubuntu 22.04 desktop. connecting over SSH. With humble installed. I followed your guides on the setup portion for connecting over USB C and ROS humble.

This is on the PI

**Reply by shamlian** [reply]
Are you using Fast-DDS or Cyclone DDS? What is the computer using (`echo $RMW_IMPLEMENTATION`) and what is the robot using (check the [webserver application page](https://iroboteducation.github.io/create3_docs/webserver/application/))?

**Reply by scottcandy34** [reply]
Cyclone DDS

**Reply by shamlian** [reply]
Your computer is running Ubuntu 22.04 desktop; what is the Pi running? And what version of ROS 2 is it running? We have not (yet) published directions for ROS 2 Humble with Ubuntu 22.04 on Raspberry Pi. I am asking all these annoying questions because I have a Pi running 22.04 (server), and a robot running H.1.0 and also Cyclone on both, but they're communicating via Wi-Fi, and I have heard that the process of getting Ubuntu 22's usb0 network connection to work is different than Ubuntu 20, and I am wondering if maybe that's the problem here, though the fact that you said that all of this works with nothing changed except the robot's version gives me pause.

**Reply by scottcandy34** [reply]
no PC just PI. the PI is running ubuntu 22.04 desktop over usbc

**Reply by scottcandy34** [reply]
The setup for the USB-C connection is good it notices it as a network and can view all ROS topics over it when running any other version. It is strange lol

**Reply by shamlian** [reply]
I agree, that's bizarre, and we will try to reproduce it here.

**Reply by shamlian** [reply]
Do you get the same results in H.1.0 running `ros2 topic list --no-daemon --spin-time 10` instead of `ros2 topic list` as @alsora suggested?

**Reply by scottcandy34** [reply]
I have not directly tested that no give me a sec

**Reply by scottcandy34** [reply]
Ok so I didn't run that yet. and just did the basic. I reinstalled the latest firmware once again. (4th time) and it seems to be ok it returns correctly and the teleop controller works great now. Such a strange issue cant say its been fun with this update lol.

**Reply by scottcandy34** [reply]
P.S. nothing has changed

**Reply by scottcandy34** [reply]
although this time I went from H.0.0 to H.1.0 and not G.4.1 to H.1.0

**Reply by kwujciak** [reply]
In my experience, partial topic list has always meant sub-par connection between my machine and the robot (whether that be pi to robot, robot to linux via wifi, etc).

**Reply by scottcandy34** [reply]
@kwujciak That would be my thoughts too but unfortunately I have tried reboots and checked network connections. But like I had mentioned above it would work by going back to galactic or H.0.0. When I did this The setup did not change once, I did not even touch the bot. It was only the last time that I got it to work. Can't say exactly what is was but its working now. Nothing was changed but the firmware.
